_search-plugins/star-tree-index.md

```diff
@@ -0,0 +1,190 @@
+---
+layout: default
+title: Star-tree index
+parent: Improving search performance
+nav_order: 54
+---
+
+# Star-tree index
+
+This is an experimental feature and is not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, join the discussion on the [OpenSearch forum](https://forum.opensearch.org/).    
+{: .warning}
+
+A star-tree index is a multi-field index that improves the performance of aggregations.
+
+OpenSearch will automatically use a star-tree index to optimize aggregations if the queried fields are part of star-tree index dimension fields and the aggregations are on star-tree index metric fields. No changes are required in the query syntax or the request parameters.
+
+## When to use a star-tree index
+
+A star-tree index can be used to perform faster aggregations. Consider the following criteria and features when deciding to use a star-tree index:
+
+- Star-tree indexes natively support multi-field aggregations.
+- Star-tree indexes are created in real time as part of the indexing process, so the data in a star-tree will always be up to date.
+- A star-tree index consolidates data, increasing index paging efficiency and using less IO for search queries.
+
+## Limitations
+
+Star-tree indexes have the following limitations:
+
+- A star-tree index should only be enabled on indexes whose data is not updated or deleted because updates and deletions are not accounted for in a star-tree index.
+- A star-tree index can be used for aggregation queries only if the queried fields are a subset of the star-tree's dimensions and the aggregated fields are a subset of the star-tree's metrics.
+- After a star-tree index is enabled, it cannot be disabled. In order to disable a star-tree index, the data in the index must be reindexed without the star-tree mapping. Furthermore, changing a star-tree configuration will also require a reindex operation.
+- [Multi-values/array values]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/index/#arrays) are not supported.
+- Only [limited queries and aggregations](#supported-queries-and-aggregations) are supported. Support for more features will be added in future versions.
+- The cardinality of the dimensions should not be very high (as with `_id` fields). Higher cardinality leads to increased storage usage and query latency.
+
+## Star-tree index structure
+
+The following image illustrates a standard star-tree index structure.
+
+<img src="{{site.url}}{{site.baseurl}}/images/star-tree-index.png" alt="A star-tree index containing two dimensions and two metrics" width="700">
+
+Sorted and aggregated star-tree documents are backed by `doc_values` in an index. `doc_values` use the following pattern:
+
+- The values are sorted based on the order of their `ordered_dimension`. In the preceding image, the dimensions are determined by the `status` setting and then by the `port` for each status.
```

Comment:
The values across fields are primary sorted based on first field in `ordered_dimension`, secondary sorted by corresponding fields mentioned in the `ordered_dimension`

Reply:
Not exact wording but we need to call out that the sort is based on the fields specified in the ordered dimension.